Upstream-Status: Inappropriate [this patch is 1.13 specific and 2.0 will be coming.]

Running PowerTop in Poky in dump mode (with --dump) always ends with a
segmentation fault.

The cause is: in dump mode, setup_windows() is not invoked so the
*_window variables, like cstate_window, are still NULL; later in
main() -> show_cstates(), the functions, like wrefresh(), will trigger
segfault.

After discussing in PowerTop mailling list, I think the right solution is
we should stub out the ncurses calls in dump mode.

(The patch was also sent to upstream PowerTop mailling list.)

Signed-off-by Dexuan Cui <dexuan.cui@intel.com>, 2010-08-20
(this patch is licensed under GPLv2)

--- a/display.c
+++ b/display.c
@@ -44,6 +44,12 @@
 static WINDOW *status_bar_window;
 
 #define print(win, y, x, fmt, args...) do { if (dump) printf(fmt, ## args); else mvwprintw(win, y, x, fmt, ## args); } while (0)
+#define wattrset(win, c) do { if (!dump ) wattrset(win, c); } while (0)
+#define wbkgd(win, c) do { if (!dump ) wbkgd(win, c); } while (0)
+#define werase(win) do { if (!dump ) werase(win); } while (0)
+#define wrefresh(win) do { if (!dump ) wrefresh(win); } while (0)
+#define wattron(win, a) do { if (!dump ) wattron(win, a); } while (0)
+#define wattroff(win, a) do { if (!dump ) wattroff(win, a); } while (0)
 
 char status_bar_slots[10][40];
 
